.changeset/fair-knives-relax.md

```diff
@@ -0,0 +1,6 @@
+---
+'@backstage/core': patch
+'@backstage/core-api': patch
+---
+
+Catch unhandled promise rejections & forward them to the ErrorApi to align with general error handling
```

Comment:
Probably mention how we're introducing a new `UnhandledErrorForwarder` export and how it's installed by default. And if users wish to omit that functionality they can override the API in their app.